Sources/LanguageServerProtocol/Notifications/WorkDoneProgress.swift

```diff
@@ -10,9 +10,22 @@
 //
 //===----------------------------------------------------------------------===//
 
-public enum WorkDoneProgress: NotificationType, Hashable {
+public struct WorkDoneProgress: NotificationType, Hashable {
   public static var method: String = "$/progress"
 
+  /// The progress token provided by the client or server.
+  public var token: ProgressToken
+
+  /// The progress data.
+  public var value: WorkDoneProgressType
+
+  public init(token: ProgressToken, value: WorkDoneProgressType) {
+    self.token = token
+    self.value = value
+  }
+}
+
+public enum WorkDoneProgressType: Codable, Hashable {
```

Comment:
Nitpick: Could you rename this to `WorkDoneProgressKind`? `Type` always makes me think of metatypes or generic types in the swift language, which this isn’t. That’s also more consistent with namings in the rest of the package.